Lib/test/test_importlib/extension/test_loader.py

```diff
@@ -145,6 +146,15 @@ def test_reload(self):
             importlib.reload(module)
             self.assertIs(ex_class, module.Example)
 
+    def test_reload_faliure(self):
+        '''Test that reload throws ModuleNotFounderror if reloading'''
+        '''a module whose spec cannot be found'''
+        datetime.__name__ = 'destroyed_datetime'
```

Comment:
All the helpers for importlib-related tests can be found in https://github.com/python/cpython/blob/master/Lib/test/test_importlib/util.py. In this instance you probably want both uncache() and temp_module().